new book components, unplugged tab on homepage

danipop1 opened the pull request:
TextBoxImage.tsx - new component to enter text into a text box corresponding with an image
DragAndDrop.tsx - Upon dragging and dropping an image into another, a completion message is displayed
ImageClickAnimation.tsx - click an action button to make an image perform an action (spin, jump, freeze)
CountdownTimer.tsx - timer that counts down from a default of 10 seconds

Added "UNPLUGGED" to categories, so that future unplugged books can be properly sorted and rendered

***double check that I didn't mess up anything in requirements.txt on the backend...I was trying to run it on my machine but I thought i reverted all changes once I realized it wasn't necessary***

scout517 commented:
1. The requirements.txt files look the same. Delete the one with .txt~
2. The books from the previously selected category show when clicking the unplugged category.
3. Revert back the change you made in HomePage.tsx. If the selected category is null, then it should show all the books. With the change you made you can't deselect a category by clicking it again.
4. Remove ImageClickAnimationWrong . wrongTrickMode can be passed as props to ImageClickAnimation.
5. Can you move all your images to the bucket storage? A link to it is in the Trackingsheet with the login information.
6. What is the purpose of the countdown timer component?

danipop1 commented:
See that 1-4 have been addressed properly. 
5. I moved my images to the bucket storage, and deleted the public image folder. 
6. The purpose of the countdown timer is part of my last component, its to "put your pet in timeout" to illustrate the concept of negative reinforcement (this is to aid in the explanation of AI reinforcement learning.

scout517 commented:
You can completely remove the ImageClickAnimationWrong one from BookImage.tsx and compnentEditorDefault.ts. You can pass in true or false for wrongTrickMode in the book editor.

The image URL for ImageClickAnimation in componentEditorDefault is wrong. It should be https://codekids-minio.endeavour.cs.vt.edu/codekids/PetTrainingBook/dog-clipart-boy-3.png

Clicking Unplugged still loads all books. It has to do with the backend schema, which I can fix myself.

scout517 reviewed (approved):
Everything looks good!
